### PR TITLE
Remove kotlin dependencies

### PR DESCRIPTION
**Pull request by kerkhofsd**
Remove
- usage of Kotlin classes (e.g. Intrinsics)
- Kotlin dependencies

### CODE REVIEW

File: annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/actions/ParameterMapping.java [hunk 1]
```diff
@@ -19,11 +18,17 @@
...
+        if (parameterDefinition == null) {
+            throw new IllegalArgumentException("parameterDefinition is null");
+        }
 
         this.index = index;
         String parameterDefinitionName = parameterDefinition.getName();
-        Intrinsics.checkExpressionValueIsNotNull(parameterDefinitionName, "parameterDefinition.name");
+
+        if (parameterDefinitionName == null) {
+            throw new IllegalStateException("parameterDefinitionName is null");
```
Comment (tgeens) [inline annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/actions/ParameterMapping.java]: I would still throw an `IllegalArgumentException` here

File: annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/policy/BehaviourProxy.java [hunk 2]
```diff
@@ -57,15 +60,19 @@ public BehaviourProxy(@NotNull Behaviour behaviour, @NotNull Timer timer) {
                         this.timer);
                 Object proxy = Proxy
                         .newProxyInstance(this.getClass().getClassLoader(), new Class[]{policy}, proxyHandler);
-                Intrinsics.checkExpressionValueIsNotNull(proxy, "proxy");
+                if (proxy == null) {
```
Comment (tgeens) [inline annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/policy/BehaviourProxy.java]: `Proxy.newProxyInstance(...)` never returns `null` anyway

https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Proxy.html#newProxyInstance-java.lang.ClassLoader-java.lang.Class:A-java.lang.reflect.InvocationHandler-

File: blueprint-integration/shared/src/main/java/com/github/dynamicextensionsalfresco/blueprint/DynamicExtensionsApplicationContext.java [hunk 3]
```diff
@@ -156,7 +157,7 @@ protected void loadBeanDefinitions(@NotNull DefaultListableBeanFactory beanFacto
     @Override
     protected void initBeanDefinitionReader(@Nullable XmlBeanDefinitionReader beanDefinitionReader) {
         if (beanDefinitionReader == null) {
-            Intrinsics.throwNpe();
+            throw new IllegalArgumentException("beanDefinitionReader is null");
```
Comment (tgeens) [inline blueprint-integration/shared/src/main/java/com/github/dynamicextensionsalfresco/blueprint/DynamicExtensionsApplicationContext.java]: The annotations says `beanDefinitionReader` can be `null` ?!

Comment (kerkhofsd) [reply blueprint-integration/shared/src/main/java/com/github/dynamicextensionsalfresco/blueprint/DynamicExtensionsApplicationContext.java]: This contradiction was already present in the original Kotlin code:
`    
override fun initBeanDefinitionReader(beanDefinitionReader: XmlBeanDefinitionReader?) {
        beanDefinitionReader!!.resourceLoader = this
        beanDefinitionReader.setNamespaceHandlerResolver(
                CompositeNamespaceHandlerResolver(
                        getOsgiNamespaceHandlerResolver(),
                        DefaultNamespaceHandlerResolver(classLoader
                ), getHostNamespaceHandlerResolver())
        )
        beanDefinitionReader.setEntityResolver(CompositeEntityResolver(getOsgiEntityResolver(), DelegatingEntityResolver(classLoader), getHostEntityResolver()))
    }
`
But it indeed doesn't make any sense.

**Pull request merged**